cypress/support/step_definitions/fireboltCalls.js

```diff
@@ -108,105 +108,147 @@ Given(/1st party app invokes the (?:'(.+)' )?API to '(.+)'$/, async (sdk, key) =
-  // Fetching the data like method, param, context and action etc.
-  cy.fireboltDataParser(key, sdk).then((parsedDataArr) => {
-    parsedDataArr.forEach((parsedData) => {
-      // If appId is having '3rd party app' taking default appId else using the value as-is.
-      appId = !appId
+Given(
+  /'(.+)' invokes the '(.+)' API to '(.+)'(?: on '(.+)' device)?$/,
+  async (appId, sdk, key, deviceIdentifier) => {
+    // If appId is having '3rd party app' taking default appId else using the value as-is.
+    appId = !appId
+      ? UTILS.getEnvVariable(CONSTANTS.THIRD_PARTY_APP_ID)
+      : appId === CONSTANTS.THIRD_PARTY_APP
         ? UTILS.getEnvVariable(CONSTANTS.THIRD_PARTY_APP_ID)
-        : appId === CONSTANTS.THIRD_PARTY_APP
-          ? UTILS.getEnvVariable(CONSTANTS.THIRD_PARTY_APP_ID)
-          : UTILS.checkForSecondaryAppId(appId);
-      const method = parsedData.method;
-      const param = parsedData.params;
-      const context = parsedData.context;
-      const action = parsedData.action;
-      const expected = parsedData.expected;
-      let isNotSupportedApi = false;
+        : UTILS.checkForSecondaryAppId(appId);
+    let requestTopic, responseTopic;
+    const deviceName = deviceIdentifier;
 
-      if (UTILS.isScenarioExempted(method, param)) {
-        isNotSupportedApi = true;
-      }
+    // Unable to find the ${envAppIdKey} value in the env, please add the value in configModule/constants/config.json
+    cy.then(() => {
       if (
-        Cypress.env(CONSTANTS.TEST_TYPE) &&
-        Cypress.env(CONSTANTS.TEST_TYPE).toLowerCase() == CONSTANTS.MODULE_NAMES.LIFECYCLE
+        deviceIdentifier &&
+        !UTILS.getEnvVariable(deviceIdentifier, false) &&
+        deviceIdentifier != CONSTANTS.DEVICE1
       ) {
-        cy.fetchLifecycleHistory(appId);
-      }
-      const communicationMode = UTILS.getCommunicationMode();
-      const additionalParams = {
-        communicationMode: communicationMode,
-        action: action,
-        isNotSupportedApi: isNotSupportedApi,
-      };
-      const params = { method: method, methodParams: param };
-
-      // Creating intent message using above details to send it to 3rd party app.
-      const intentMessage = UTILS.createIntentMessage(
-        CONSTANTS.TASK.CALLMETHOD,
-        params,
-        additionalParams
-      );
-
-      fireLog.info(`Call from ${appId}, method: ${method} params: ${JSON.stringify(param)}`);
-      if (Cypress.env('isRpcOnlyValidation')) {
-        fireLog.info(
-          `${method} response will be retrieved in subsequent steps and validated when the rpc-only methods are invoked. Proceeding to the next step.`
+        fireLog.assert(
+          false,
+          `Unable to find the ${deviceIdentifier} environment value, Verify the value of ${deviceIdentifier} in configModule or cli command.}`
         );
+      } else if (deviceIdentifier && UTILS.getEnvVariable(deviceIdentifier, false)) {
```

Comment:
Can you add comments explaining the if conditions? I'm thinking what should be happening is if deviceIdentifier is passed we lookup that device otherwise default to deviceMac, but the conditions are not obvious.

Reply:
Added the comment for this conditions also added one note saying how we are passing the device identifier